### PR TITLE
(DOCSP-26904) [BAAS] Configure user creation function (CLI/API)

### PR DESCRIPTION
**Pull request by nlarew**
### Jira

- (DOCSP-26904) [BAAS] Configure user creation function (CLI/API)

### Staged Changes

- [Enable Custom User Data > User Creation Function](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/user-creation-fn/users/enable-custom-user-data/)
- [Config > Custom User Data](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/user-creation-fn/manage-apps/configure/config/auth/#custom-user-data)
- [Admin API > Custom User Data](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/user-creation-fn/admin/api/v3/#tag/custom-user-data)

### CODE REVIEW

File: source/users/enable-custom-user-data.txt [hunk 1]
```diff
@@ -72,6 +72,54 @@ configure custom user data for your application, you specify:
...
+User Creation Function
+~~~~~~~~~~~~~~~~~~~~~~
+
+You can define a function that runs every time a new user successfully
+registers but before their new user account is created. If the function
+throws or otherwise errors, the user account creation fails. This lets
+you ensure that users always have custom data associated with them once
+created.
+
+The function receives a user metadata object as its only argument. You
```
Comment (mongodben) [inline source/users/enable-custom-user-data.txt:88]: can you link to a ref for 'user metadata object'? prob ref at top of this page https://www.mongodb.com/docs/atlas/app-services/authentication/user-objects/

File: source/users/enable-custom-user-data.txt [hunk 2]
```diff
@@ -72,6 +72,54 @@ configure custom user data for your application, you specify:
 - the user ID field used to map custom user data documents
   to users (via user ID)
 
+- an optional function that runs before new user accounts are created
```
Comment (mongodben) [inline source/users/enable-custom-user-data.txt:75]:
```suggestion
- an optional Atlas Function that runs before new user accounts are created
```

File: source/users/enable-custom-user-data.txt [hunk 3]
```diff
@@ -72,6 +72,54 @@ configure custom user data for your application, you specify:
...
+
+.. _user-creation-function:
+
+User Creation Function
+~~~~~~~~~~~~~~~~~~~~~~
+
+You can define a function that runs every time a new user successfully
```
Comment (mongodben) [inline source/users/enable-custom-user-data.txt:82]:
```suggestion
You can define a Function that runs every time a new user successfully
```

also pls capitalize for service name throughout rest of page

File: source/manage-apps/configure/config/auth.txt [hunk 4]
```diff
@@ -201,3 +202,7 @@ collection for your app in ``/auth/custom_user_data.json``.
      - The name of the field in custom user data documents that
        contains the user ID of the application user the document
        describes.
+
+   * - | ``on_user_creation_function_name``
+       | String
+     - The name of the :ref:`user creation function <user-creation-function>`.
```
Comment (mongodben) [inline source/manage-apps/configure/config/auth.txt:208]:
```suggestion
     - The name of the :ref:`user creation Atlas Function <user-creation-function>`.
```

File: source/openapi-admin-v3.yaml [hunk 5]
```diff
@@ -4661,6 +4701,45 @@ components:
...
+          description: |-
+            The name of a field that appears in every custom user data
+            document and that contains the user account ID of each
+            document's corresponding user.
+          example: "user_account_id"
+        on_user_creation_function_id:
+          type: string
+          description: |-
+            The `_id` value of a [user creation
+            function](https://www.mongodb.com/docs/atlas/app-services/users/enable-custom-user-data/#std-label-user-creation-function)
```
Comment (mongodben) [inline source/openapi-admin-v3.yaml:4736]:
```suggestion
            The `_id` value of a [user creation
            Atlas Function](https://www.mongodb.com/docs/atlas/app-services/users/enable-custom-user-data/#std-label-user-creation-function)
```

**Review by mongodben — APPROVED** [review]
lgtm w some small buts and bobs throughout to fix

File: source/manage-apps/configure/config/auth.txt (hunk 4 above)
Comment (nlarew) [reply source/manage-apps/configure/config/auth.txt:208]: gonna leave these - "user creation function" doesn't need to reference the service name in the same way that a "trigger function" isn't a "Trigger Atlas Function"

**Pull request merged**